src/Tasks/Microsoft.Common.CurrentVersion.targets

```diff
@@ -1688,9 +1688,22 @@ Copyright (C) Microsoft Corporation. All rights reserved.
 
       <!--
          Remove the items we've touched from _MSBuildProjectReferenceExistent. This will leave all projects where
-         SkipGetTargetFrameworkProperties was set. Then add all AnnotatedProjects back.
+         SkipGetTargetFrameworkProperties was set.
       -->
       <_MSBuildProjectReferenceExistent Remove="@(_MSBuildProjectReferenceExistent)" Condition="'%(_MSBuildProjectReferenceExistent.SkipGetTargetFrameworkProperties)' != 'true'" />
+
+      <!-- If we skipped querying the other project, ensure we don't pollute it with the TargetFramework/RuntimeIdentifier
+           that may be set for this project as a global property (if we're in an inner build).
+
+           But allow setting TF explicitly. -->
+      <_MSBuildProjectReferenceExistent Condition="'%(_MSBuildProjectReferenceExistent.SetTargetFramework)' == ''">
+          <UndefineProperties>%(_MSBuildProjectReferenceExistent.UndefineProperties);TargetFramework</UndefineProperties>
+      </_MSBuildProjectReferenceExistent>
+      <_MSBuildProjectReferenceExistent>
+          <UndefineProperties>%(_MSBuildProjectReferenceExistent.UndefineProperties);RuntimeIdentifier</UndefineProperties>
```

Comment:
@nguerrera Do you think we should clear RID when traversing to a project that had `SkipGetTargetFrameworkProperties=true`? We weren't before. Maybe I should just leave it out here too?

Reply:
We need to run this through the SDK tests as we did for the original double eval fix. I forget all of the subtleties of RID cases that @TheRealPiotrP worked through.

Reply:
cc @dsplaisted His recent work might make running those tests against an arbitrary msbuild  (or at least CLI) easier.

Reply:
That'd be exciting! It would be nice to see if we could get an optional Jenkins test to run SDK tests on new MSBuild bits (long term . . .)